caffe2/serialize/file_adapter.cc

```diff
@@ -40,16 +38,17 @@ FileAdapter::RAIIFile::~RAIIFile() {
 
 // FileAdapter directly calls C file API.
 FileAdapter::FileAdapter(const std::string& file_name) : file_(file_name) {
-  const int fseek_ret = fseek(file_.fp_, 0L, SEEK_END);
+  auto fseek_ret = fseek(file_.fp_, 0L, SEEK_END);
   TORCH_CHECK(fseek_ret == 0, "fseek returned ", fseek_ret);
 #if defined(_MSC_VER)
-  const int64_t ftell_ret = _ftelli64(file_.fp_);
+  const auto ftell_ret = _ftelli64(file_.fp_);
 #else
-  const off_t ftell_ret = ftello(file_.fp_);
+  const auto ftell_ret = ftello(file_.fp_);
 #endif
   TORCH_CHECK(ftell_ret != -1L, "ftell returned ", ftell_ret);
   size_ = ftell_ret;
-  rewind(file_.fp_);
+  fseek_ret = fseek(file_.fp_, 0L, SEEK_SET);
```

Comment:
Why?

Reply:
rewind has no return value to indicate system errors. The man page recommends to use fseek in this way.

Reply:
Did we add an assertion check or something?